src/inspector_agent.cc

```diff
@@ -149,8 +155,165 @@ static int RegisterDebugSignalHandler() {
+    TryCatch try_catch(isolate);
+    Local<Function> callback = callback_.Get(isolate);
+    Local<Object> receiver = receiver_.Get(isolate);
+    callback->Call(receiver, 1, &argument);
+    if (try_catch.HasCaught()) {
+      node::Utf8Value error_string(isolate, try_catch.Message()->Get());
+      fprintf(stderr, "Exception in the inspector message handler: %s\n",
+              error_string.out());
+    }
+  }
+
+  void Disconnect() {
+    Agent* agent = env_->inspector_agent();
+    if (agent->delegate() == this)
+      agent->Disconnect();
+  }
+
+ private:
+  static void Release(
+      const v8::WeakCallbackInfo<JsBindingsSessionDelegate>& info) {
+    info.SetSecondPassCallback(ReleaseSecondPass);
+    info.GetParameter()->session_.Reset();
+  }
+
+  static void ReleaseSecondPass(
+      const v8::WeakCallbackInfo<JsBindingsSessionDelegate>& info) {
+    JsBindingsSessionDelegate* delegate = info.GetParameter();
+    delegate->Disconnect();
+    delete delegate;
+  }
+  Environment* env_;
+  Persistent<Object> session_;
+  Persistent<Object> receiver_;
+  Persistent<Function> callback_;
+};
+
+Maybe<Agent*> GetInspectorChecked(Environment* env) {
+  Agent* agent = env->inspector_agent();
+  if (agent == nullptr) {
+    env->ThrowError("Inspector was not setup");
+    return v8::Nothing<Agent*>();
+  }
+  return v8::Just(agent);
+}
+
+void SetDelegate(Environment* env, Local<Object> inspector,
+                 JsBindingsSessionDelegate* delegate) {
+  inspector->SetPrivate(env->context(),
+                        env->inspector_delegate_private_symbol(),
+                        v8::External::New(env->isolate(), delegate));
+}
+
+Maybe<JsBindingsSessionDelegate*> GetDelegate(Environment* env,
+                                              Local<Object> inspector) {
+  Local<Value> delegate;
+  MaybeLocal<Value> delegateField =
+      inspector->GetPrivate(env->context(),
+                            env->inspector_delegate_private_symbol());
+
+  if (!delegateField.ToLocal(&delegate))
+    return v8::Nothing<JsBindingsSessionDelegate*>();
+  CHECK(delegate->IsExternal());
+  void* value = delegate.As<External>()->Value();
+  if (value == nullptr)
+    return v8::Nothing<JsBindingsSessionDelegate*>();
+  return v8::Just(static_cast<JsBindingsSessionDelegate*>(value));
+}
+
+void Dispatch(const FunctionCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  if (!info[0]->IsString()) {
+    env->ThrowError("Inspector message must be a string");
+    return;
+  }
+  Maybe<Agent*> maybe_inspector = GetInspectorChecked(env);
+  if (maybe_inspector.IsNothing())
+    return;
+  Agent* inspector = maybe_inspector.ToChecked();
+  InspectorSessionDelegate* delegate = inspector->delegate();
+  if (delegate == nullptr
+      || delegate != GetDelegate(env, info.This()).FromMaybe(nullptr)) {
+    env->ThrowError("Inspector is not connected");
+    return;
+  }
```

Comment:
I redid the code. Original version was written before the signal handler was merged in. Now V8 inspector is always present and there's no need to check if it exists.